lms/product/canvas/_plugin/course_service.py

```diff
@@ -0,0 +1,31 @@
+from lms.models import ApplicationSettings, CourseGroupsExportedFromH
+from lms.product.plugin.course_service import CourseServicePlugin
+
+
+class CanvasCoursePlugin(CourseServicePlugin):
+    def __init__(self, db_session, parsed_params):
+        self._db_session = db_session
+        self._parsed_params = parsed_params
+
+    def get_new_course_settings(
+        self, settings: ApplicationSettings, authority_provided_id
+    ) -> ApplicationSettings:
+        # Disable sections for courses which existed before sections but we
+        # haven't seen since
+        if self._db_session.query(CourseGroupsExportedFromH).get(authority_provided_id):
+            settings.set("canvas", "sections_enabled", False)
```

Comment:
This is actually a change in behavior. Previously this did not only apply in the case of Canvas, but now it will. This doesn't seem like it should logically be a problem, but who knows?

Also over time, this seems less and less likely that it matters at all. As courses this old we've never seen since they were created, and are still being re-used should get rarer over time.